Don't allow CREATE TABLE with quicklz, if server is built without quicklz.

hlinnaka opened the pull request:
Commit 46b21685f7 added a similar check for Zstandard. This makes QuickLZ
behave the same.

Fixes https://github.com/greenplum-db/gpdb/issues/5107

danielgustafsson commented on src/backend/access/common/reloptions_gp.c:
```diff
@@ -964,16 +964,24 @@ validate_and_adjust_options(StdRdOptions *result,
+				if (validate)
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							 errmsg("compresslevel=%d is out of range for quicklz (should be 1)",
```
s/quicklz/QuickLZ/ for consistency?

hlinnaka replied:
Hmm. The Zstandard messages have the same pattern. Yeah, might be good to reword these, but I'l leave that to a separate PR.

berlin-ab commented on src/backend/access/common/reloptions_gp.c:
```diff
@@ -964,16 +964,24 @@ validate_and_adjust_options(StdRdOptions *result,
+				if (validate)
+					ereport(ERROR,
+							(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							 errmsg("compresslevel=%d is out of range for quicklz (should be 1)",
```
I'm surprised that "zstd" wasn't a constant somewhere. The magic numbers of `19` (for zstd) and `1` (for quicklz) should be `#define`d somewhere.

hlinnaka replied:
Agreed. I'm not going to try fixing that as part of this PR, though.

ivannovick commented:
This makes TONS of sense, thx!

hlinnaka commented:
Rebased, and added another HAVE_LIBQUICKLZ check. I didn't notice earlier that there was a second copy of essentially the same code that also needs this.

berlin-ab commented on src/backend/access/common/reloptions_gp.c:
```diff
@@ -1185,13 +1193,19 @@ validateAppendOnlyRelOptions(bool ao,
+					 errmsg("QuickLZ library is not supported by this build"),
+					 errhint("Compile with --with-quicklz to use QuickLZ compression.")));
+#endif
+			if (complevel != 1)
```
More evidence that we should create a #define for this expected compression level number.
